content/06.discussion.md

```diff
@@ -54,6 +54,54 @@ In assessing the upper bound on the predictive performance of a deep learning mo
+Hence, it is crucial to obtain uncertainty measures that capture how noise in input values propagate through deep neural networks.
+Such measures can be used for reliability assessment of automated decisions in clinical and public health applications, and for guarding against model vulnerabilities in the face of rare or adversarial cases [@tag:ghahramani_protect].
+Moreover, in fundamental biological research, measures of uncertainty help researchers distinguish between true regularities in the data and patterns that are false or merely anecdotal.
+There are two main uncertainties that one can calculate: epistemic uncertainty, which describes uncertainty about the model and is caused by insufficient training data or by a difference in the training set and testing set distributions (this uncertainty vanishes in the limit of infinite data), and aleatoric uncertainty, which describes uncertainty inherent in the observations and is due to noisy or missing data (this uncertainty vanishes with the ability to observe all independent variables with infinite precision) [@tag:uncertainty_types].
+Aleatoric uncertainty can be represented well by designing an appropriate loss function.
+In the case of data-dependent aleatoric uncertainty, one can train the model to increase its uncertainty when it is incorrect due to noisy or missing data, and in the case of task-depedent aleatoric uncertainty, one can optimize for the best uncertainty parameter for each task [@tag:uncertainty_multi_task].
+Meanwhile, there are various methods for modeling epistemic uncertainty, which we outline in what follows.
+
+In classification tasks, confidence calibration is the problem of using classifier scores to predict class membership probabilities that match the true membership likelihoods.
+These membership probabilities can be used to assess the uncertainty associated with assigning the example to each of the classes.
+Guo et al. [@tag:guo_calibration] observed that contemporary neural networks are poorly calibrated and provided a simple recommendation for calibration: temperature scaling, a single parameter special case of Platt scaling [@tag:platt_scaling].
+In addition to confidence calibration, there is early work from Chryssolouris et al. [@tag:Chryssolouris1996_confidence] that described a method for obtaining confidence intervals with the assumption of normally distributed error for the neural network.
+More recently, Hendrycks and Gimpel discovered that incorrect or out-of-distribution examples usually have lower maximum softmax probabilities than correctly classified examples, allowing for effective detection of misclassified examples [@tag:out_dist_baseline].
+Liang et al. used temperature scaling and small perturbations to further separate the softmax scores of correctly classified examples and the scores of out-of-distribution examples, allowing for more effective detection [@tag:temp_out_dist].
```

Comment:
Can you really tie this paragraph together with one short sentence that provides a recommendation on a current best approach? Sounds like Liang et al. combines the two, so if that does provide the current best of both worlds, maybe say that at the conclusion of that sentence.

Reply:
Added a conclusion to this paragraph.